homeassistant/components/bang_olufsen/websocket.py

```diff
@@ -92,8 +92,14 @@ def on_notification_notification(
         self, notification: WebsocketNotificationTag
     ) -> None:
         """Send notification dispatch."""
+
         if notification.value:
-            if WebsocketNotification.REMOTE_MENU_CHANGED in notification.value:
+            if WebsocketNotification.BEOLINK.value in notification.value:
+                async_dispatcher_send(
+                    self.hass,
+                    f"{self._unique_id}_{WebsocketNotification.BEOLINK}",
+                )
+            elif notification.value is WebsocketNotification.REMOTE_MENU_CHANGED.value:
```

Comment:
Why is this condition different than the old comparison?

Reply:
The `WebsocketNotification.CONFIGURATION` notification dispatch was removed because the the Home Assistant device name is no longer updated when the device's name is updated. This was necessary to have up to date Beolink extra state attributes, where the name and JID was shown.

Reply:
One comparison uses "is" the other "in". Just want to double check that's correct.   Codecov reporting some gaps here also.

Reply:
Ah I see, yes that is on purpose. There is no "beolink" notification, but several notifications that contain "beolink", such as: "beolinkPeers", "beolinkListeners" and "beolinkAvailableListeners", which all should trigger a beolink dispatch. Also the websocket.py file is not tested, so coverage shouldn't matter(?)

Reply:
I think it's quite weird to use an enum item for a substring check, that's not how I'd expect an enum to be used.

As a rule of thumb, integration tests should as a minimum mock the API used by the integration and use service calls and the state machine to test the integration.

This is not the way the B&O tests are implemented though considering `websocket.py` consumes callbacks from the B&O and forwards them to the entities via signals. The tests should capture the websocket callback registrations done by `BangOlufsenWebsocket` and call those callbacks instead of directly sending signals.

Can you please look into that? I think it's a relatively small change, which has the nice benefit of indirectly covering websocket.py.

Reply:
I've opened a PR for the proper use and testing of websocket.py: https://github.com/home-assistant/core/pull/123075

Reply:
The updated code is much clearer.  Thank you.